runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ControlClientPool.java

```diff
@@ -17,16 +17,12 @@
  */
 package org.apache.beam.runners.fnexecution.control;
 
-import org.apache.beam.sdk.fn.function.ThrowingConsumer;
-import org.apache.beam.sdk.util.ThrowingSupplier;
-
 /** Control client pool that exposes a source and sink of control clients. */
-public interface ControlClientPool<T extends InstructionRequestHandler> {
+public interface ControlClientPool {
 
   /** Source of control clients. */
```

Comment:
<!--new_thread; commit:1fd121da1417624b3b84f0300648251da64b9cb5; resolved:0-->
Generally, why the higher-order functions here: why not have ControlClientPool have a function for getting a client, and a function for adding a client to the pool? That would match how other pool-like abstractions typically work.